src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java

```diff
@@ -0,0 +1,31 @@
+package com.wolox.training.exceptions;
+
+import javax.validation.ConstraintViolationException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler({ BookNotFoundException.class })
+    protected ResponseEntity<Object> handleNotFound(
+        Exception ex, WebRequest request) {
+        return handleExceptionInternal(ex, "Book not found",
+            new HttpHeaders(), HttpStatus.NOT_FOUND, request);
+    }
+
+    @ExceptionHandler({ BookIdMismatchException.class,
+        ConstraintViolationException.class,
+        DataIntegrityViolationException.class })
+    public ResponseEntity<Object> handleBadRequest(
+        Exception ex, WebRequest request) {
+        return handleExceptionInternal(ex, ex.getLocalizedMessage(),
```

Comment:
Is not correct return the generar error, add one more readable

Reply:
so, in this one you recommend something like `Book mismatch` or something like that?

Reply:
Yes, i like that